### PR TITLE
Fixed filter error for index scan

### PR DESCRIPTION
**Pull request by bright-starry-sky**

### CODE REVIEW

File: src/storage/index/IndexPolicyMaker.cpp [hunk 1]
```diff
@@ -78,6 +78,12 @@ bool IndexPolicyMaker::buildPolicy() {
         if (exist == scanItems_.end()) {
             break;
         }
+        // Stop build policy when last operator is range scan,
+        // And other fields will use expression filtering.
+        auto it = scanItems_.rbegin()->second;
```
Comment (critical27) [inline src/storage/index/IndexPolicyMaker.cpp:83]: Is `rbegin` wrong? We can't make sure that the last index column is the `rbegin` of `scanItems`.

**Comment by critical27** [comment]
index (c1, c2 ,c3), where c1 == 1 and c2 > 1 and c2 < 3 and c3 == 4, [begin, end) is [1_2, 1_3)

File: src/storage/index/IndexPolicyMaker.cpp (hunk 1 above)
Comment (bright-starry-sky) [reply src/storage/index/IndexPolicyMaker.cpp:83]: > Is `rbegin` correct? We can't make sure that the last index column is the `rbegin` of `scanItems`.

I think this is correct. we just have to make sure it's the last one in scanItems_, not index fields. WDYT?

**Comment by bright-starry-sky** [comment]
> index (c1, c2 ,c3), where c1 == 1 and c2 > 1 and c2 < 3 and c3 == 4, [begin, end) is [1_2, 1_3)

yeah.

File: src/storage/index/IndexPolicyMaker.cpp (hunk 1 above)
Comment (critical27) [reply src/storage/index/IndexPolicyMaker.cpp:83]: But the scanItems_ is a map<index_column_name, ScanBound>? The `rbegin` is the last element of a map, not the column name in index order?

Comment (bright-starry-sky) [reply src/storage/index/IndexPolicyMaker.cpp:83]: > But the scanItems_ is a map<index_column_name, ScanBound>? The `rbegin` is the last element of a map, not the column name in index order?

you are right. my bad.

**Comment by bright-starry-sky** [comment]
Addressed critical27's comment. Thanks.

**Review by critical27 — APPROVED** [review]
LGTM

**Review by panda-sheep — APPROVED** [review]
Well done！

**Pull request merged**